R/DataWrangling.R

```diff
@@ -285,3 +285,51 @@ FindCovariateNames <- function(df) {
 GetFriendlyCovariateName <- function(column_name) {
   return(stringr::str_replace(column_name, .covariate_prefix_regex, ""))
 }
+
+
+
+#' Keep or delete rows in @param data corresponding to the control treatment in each study.
+#' 
+#' @param data Data in long format, plus the column 'Treatment', a text version of 'T'.
+#' @param treatments Vector of treatments with the reference treatment first.
+#' @param keep_delete "keep" or "delete".
+#' @return @param data with rows corresponding to the control treatment kept or deleted, and a new column 'Control'.
+KeepOrDeleteControlTreatment <- function(data, treatments, keep_delete){
+  #It is imperative that 'data' is sorted by 'Study', because tapply below will sort by 'Study', creating a misalignment between 'Study' and 'Control' in the 'control' data frame if 'data' is not sorted by 'Study' already.
+  data <- dplyr::arrange(data, Study, T)
+  #The unique studies
+  studies <- unique(data$Study)
+  #Local function to find the control treatment in a single study
+  #When used in tapply it matches the treatments within a study to the ordered 'treatments' vector, and then finds the lowest
+  min_match <- function(x){min(match(x, treatments))}
+  #Find the control treatment in each study
+  control <- data.frame(Study = studies, Control = treatments[tapply(data$Treatment, INDEX = data$Study, FUN = min_match)])
+  data <- merge(data, control, by = "Study", sort = FALSE)
+  if(keep_delete == "keep"){
+    return(data[data$Treatment == data$Control, ])
+  } else if(keep_delete == "delete"){
+    return(data[data$Treatment != data$Control, ])
+  } else{stop("keep_delete must be 'keep' or 'delete'")}
+}
+
+
+
+#' Delete rows in @param data corresponding to the control treatment in each study.
```

Comment:
@JanionNevill What do you think about this? I know what I've done is a standard thing to do in software development, but I'm not sure why.

Reply:
I could go either way on this. Generally when this pattern is used, it because the intermediate functions do something which is less obvious. In this case, they just add a string which matches the name, so it's not really adding any more clarity. I think for that reason, there isn't much point having the 2 intermediate functions.

Reply:
Deleted.